### PR TITLE
WT-4556 Add a way to synchronize threads in workgen.

### PR DESCRIPTION
**Pull request by ddanderson**

### CODE REVIEW

**Comment by ddanderson** [comment]
@sueloverso, can you please review these changes?

File: bench/workgen/workgen_time.h [hunk 1]
```diff
@@ -199,3 +199,9 @@ ts_us(const timespec &ts)
 {
     return (ns_to_us(ts.tv_nsec) + sec_to_us(ts.tv_sec));
 }
+
+inline uint64_t
+secs_us(double secs)
+{
+    return (ns_to_us(secs * NSEC_PER_SEC));
```
Comment (sueloverso) [inline bench/workgen/workgen_time.h]: Why multiply to nanosecs and then divide to usecs? Why not just use `secs * USEC_PER_SEC` from `workgen_time.h`?

File: bench/workgen/runner/read_write_sync_long.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+from wiredtiger import *
+from workgen import *
+
+context = Context()
+conn_config = ""
+conn_config += ",cache_size=2GB,eviction=(threads_max=8),log=(enabled=true),session_max=250,statistics=(fast),statistics_log=(wait=1,json)"   # explicitly added
+conn = wiredtiger_open("WT_TEST", "create," + conn_config)
+s = conn.open_session("")
+
+wtperf_table_config = "key_format=S,value_format=S,type=lsm," +\
```
[the remark this answers is not in the document]
Comment (ddanderson) [reply bench/workgen/runner/read_write_sync_long.py]: That is certainly not intentional.  The wtperf.py script appears to have an error.  I will fix it here and open a ticket for wtperf.py. It certainly helps explain some of the differences we were seeing in wtperf vs workgen runs. I'll rerun one to see. Thanks for catching this!

[the remark this answers is not in the document]
Comment (sueloverso) [reply bench/workgen/runner/read_write_sync_long.py]: Ha, yes that probably does explain some of the differences we saw! Even though WT-4447 is closed, maybe post in there?

[the remark this answers is not in the document]
Comment (ddanderson) [reply bench/workgen/runner/read_write_sync_long.py]: Fixed here, and I opened WT-4606 to deal with wtperf.py .

File: bench/workgen/runner/read_write_sync_long.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
+#/usr/bin/env python
+# generated from runner/read_write_heavy.wtperf originally, then hand edited.
+import sys
+# The next lines are unneeded if this script is in the runner directory.
+sys.path.append("/mnt/data0/dda/wt/git/wt-4556-workgen-synchronize/bench/workgen/runner")
```
[the remark this answers is not in the document]
Comment (ddanderson) [reply bench/workgen/runner/read_write_sync_long.py]: fixed.

File: bench/workgen/runner/read_write_sync_short.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
+#/usr/bin/env python
+# generated from runner/read_write_heavy.wtperf originally, then hand edited.
+import sys
+# The next lines are unneeded if this script is in the runner directory.
+sys.path.append("/mnt/data0/dda/wt/git/wt-4556-workgen-synchronize/bench/workgen/runner")
```
[the remark this answers is not in the document]
Comment (ddanderson) [reply bench/workgen/runner/read_write_sync_short.py]: fixed

File: bench/workgen/workgen_time.h (hunk 1 above)
Comment (ddanderson) [reply bench/workgen/workgen_time.h]: Right!  fixed.

**Pull request merged**